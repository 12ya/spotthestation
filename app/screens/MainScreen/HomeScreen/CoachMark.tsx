import React from "react"
import { ViewStyle, View, PressableProps, TextStyle } from "react-native"
import { Button, Icon, IconTypes, Text } from "../../../components"
import { TxKeyPath } from "../../../i18n"
import { fontSizes, lineHeights, scale, typography } from "../../../theme"
import { colors } from "../../../theme/colors"
import { normalizeHeight } from "../../../utils/normalizeHeight"

export interface CoachMarkProps {
  icon?: IconTypes,
  title?: TxKeyPath,
  bodyText?: TxKeyPath,
  stage?: number,
  style?: ViewStyle,
  onPressNext?: PressableProps["onPress"]
  onPressFinish?: PressableProps["onPress"]
}

export function CoachMark({ icon, title, bodyText, stage, style, onPressNext, onPressFinish }: CoachMarkProps) {
  const renderArrow = () => {
    switch (stage) {
      case 1: return <View style={$location}><Icon icon="bigArrow" size={55} /></View>
      case 2: return <View style={$sightings}><Icon icon="bigArrow" size={55} /></View>
      case 3: return <View style={$globe}><Icon icon="bigArrow" size={55} /></View>
      case 4: return <View style={$map}><Icon icon="bigArrow" size={55} /></View>
      case 5: return <View style={$navigation}><Icon icon="bigArrow" size={55} /></View>
      default: return ''
    }
  }

  return (
    <View 
      accessible
      accessibilityLabel="coach mark"
      accessibilityHint="coach mark"
      accessibilityRole="text"
      style={[$modalBodyContainer, style]}
    >
      {renderArrow()}
      <Text text={`${stage}/5`} style={$stage} />
      <Icon icon={icon} size={44} />
      <Text tx={title} style={$title} />
      <Text tx={bodyText} style={$body} />
      {stage === 5 ? <Button
          accessible
          accessibilityLabel="finish button"
          accessibilityHint="finish coach mark"
          tx="homeScreen.coachMarks.finish"
          textStyle={$nextButtonText}
          style={$nextButton}
          pressedStyle={$nextButton}
          onPress={onPressFinish}
        /> : <View style={$buttonsContainer}>
        <Button
          accessible
          accessibilityLabel="skip button"
          accessibilityHint="skip coach mark"
          tx="homeScreen.coachMarks.skip"
          textStyle={$skipButtonText}
          style={$skipButton}
          pressedStyle={$skipButton}
          onPress={onPressFinish}
        />
        <Button
          accessible
          accessibilityLabel="next button"
          accessibilityHint="next coach mark"
          tx="homeScreen.coachMarks.next"
          textStyle={$nextButtonText}
          style={$nextButton}
          pressedStyle={$nextButton}
          onPress={onPressNext}
        />
      </View>}
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.buttonBlue,
  borderRadius: scale(16),
  alignItems: 'center',
  paddingVertical: 36,
  paddingHorizontal: 30,
  width: '100%',
}

const $buttonsContainer: ViewStyle = {
  flexDirection: 'row',
  justifyContent: 'space-between',
  width: '100%',
  alignSelf: 'flex-end'
}

const $skipButton: ViewStyle = {
  backgroundColor: 'transparent',
  borderWidth: 0,
  height: scale(56),
  minWidth: scale(140)
}

const $skipButtonText: TextStyle = {
  fontFamily: typography.primary.medium,
  fontSize: fontSizes[18],
  lineHeight: lineHeights[22],
  color: colors.palette.neutral100,
}

const $nextButton: ViewStyle = {
  height: scale(56),
  backgroundColor: colors.palette.neutral100,
  borderRadius: scale(28),
  borderWidth: 0,
  minWidth: scale(140)
}

const $location: ViewStyle = {
  position: "absolute",
  top: -scale(80),
  right: -scale(5),
}
const $sightings: ViewStyle = {
  position: "absolute",
  top: -scale(80),
}
const $globe: ViewStyle = {
  position: "absolute",
  top: -normalizeHeight(.2),
  right: 0,
  transform: [{ rotate: "-130deg" }]
}
const $map: ViewStyle = {
  position: "absolute",
  bottom: -scale(80),
  transform: [{ rotate: "180deg" }]
}
const $navigation: ViewStyle = {
  position: "absolute",
  bottom: -scale(80),
  transform: [{ rotate: "180deg" }]
}

const $nextButtonText: TextStyle = {
  fontFamily: typography.primary.medium,
  fontSize: fontSizes[18],
  lineHeight: lineHeights[22],
  color: colors.palette.buttonBlue,
}

const $title: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: fontSizes[24],
  lineHeight: lineHeights[29],
  color: colors.palette.neutral100,
  paddingBottom: 12,
  paddingTop: 18
}

const $body: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: fontSizes[18],
  lineHeight: lineHeights[22],
  color: colors.palette.neutral100,
  paddingBottom: 36
}

const $stage: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: fontSizes[16],
  lineHeight: lineHeights[20],
  color: colors.palette.neutral100,
  position: "absolute",
  top: 0,
  left: 0,
  padding: 24,
  opacity: 0.5
}
