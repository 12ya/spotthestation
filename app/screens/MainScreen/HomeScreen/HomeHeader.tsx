import { StyleFn, useStyles } from "../../../utils/useStyles"
import React from "react"
import { ViewStyle, View, TextStyle, PressableProps, Pressable } from "react-native"
import { colors, typography } from "../../../theme"
import { Text } from "../../../components"
import { IconLinkButton } from "../../OnboardingScreen/components/IconLinkButton"
import { translate } from "../../../i18n"

interface User {
  firstName: string
  address: string
}

export interface HomeHeaderProps {
  user: User
  sighting?: string
  countdown?: string
  timezone?: string
  onLocationPress: PressableProps["onPress"]
  onSightingsPress: PressableProps["onPress"]
}

export function HomeHeader({
  user,
  onLocationPress,
  onSightingsPress,
  sighting = "",
  countdown = "",
  timezone = "",
}: HomeHeaderProps) {
  const {
    $headerContainer,
    $rowContainer,
    $userContainer,
    $addressText,
    $timeContainer,
    $outlined,
    $headText,
    $timeText,
    $tipText,
    $firstHeadText,
  } = useStyles(styles)

  return (
    <View style={$headerContainer}>
      <View style={$rowContainer}>
        <View style={$userContainer}>
          <Text
            accessible
            accessibilityLabel="address"
            accessibilityHint="address"
            accessibilityRole="text"
            text={user?.address?.replace(", ", "\n")}
            style={$addressText}
            ellipsizeMode="tail"
            numberOfLines={2}
          />
        </View>
        <IconLinkButton
          accessible
          accessibilityLabel="pin button"
          accessibilityHint="open select location modal"
          icon="pin"
          onPress={onLocationPress}
        />
      </View>
      <View style={$rowContainer}>
        <Pressable
          accessible
          accessibilityLabel="next sighting"
          accessibilityHint="open sightings modal"
          accessibilityRole="button"
          style={[$timeContainer, $outlined]}
          onPress={onSightingsPress}
        >
          <Text
            accessible
            accessibilityLabel="sighting header"
            accessibilityHint="sighting header"
            accessibilityRole="text"
            tx="homeScreen.header.firstTimeHead"
            style={$firstHeadText}
          />
          <Text
            accessible
            accessibilityLabel="sighting"
            accessibilityHint="sighting"
            accessibilityRole="text"
            text={sighting}
            style={$timeText}
          />
          <Text
            accessible
            accessibilityLabel="timezone"
            accessibilityHint="timezone"
            accessibilityRole="text"
            text={`${translate("homeScreen.header.timezone")}: ${timezone}`}
            style={$tipText}
          />
        </Pressable>
        <View
          accessible
          accessibilityLabel="countdown"
          accessibilityHint="countdown to next sighting"
          accessibilityRole="text"
          style={$timeContainer}
        >
          <Text tx="homeScreen.header.secondTimeHead" style={$headText} />
          <Text text={countdown} style={$timeText} />
          <Text text="DD:HH:MM:SS" style={$tipText} />
        </View>
      </View>
    </View>
  )
}

const styles: StyleFn = ({ scale, fontSizes, lineHeights }) => {
  const $headerContainer: ViewStyle = {
    width: "100%",
    paddingHorizontal: 18,
    paddingTop: 18,
  }

  const $rowContainer: ViewStyle = {
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "space-between",
    paddingBottom: 5,
  }

  const $userContainer: ViewStyle = {
    width: "85%",
  }

  const $addressText: TextStyle = {
    fontFamily: typography.primary.normal,
    color: colors.palette.neutral100,
    fontSize: fontSizes[20],
    lineHeight: lineHeights[26],
  }

  const $timeContainer: ViewStyle = {
    backgroundColor: colors.palette.neutral350,
    width: "48%",
    borderRadius: scale(10),
    paddingVertical: 8,
    borderWidth: scale(1.5),
  }

  const $outlined: ViewStyle = {
    borderColor: colors.palette.buttonBlue,
  }

  const $headText: TextStyle = {
    fontFamily: typography.primary.normal,
    color: colors.palette.neutral450,
    fontSize: fontSizes[14],
    lineHeight: lineHeights[16],
    textAlign: "center",
  }

  const $timeText: TextStyle = {
    fontFamily: typography.primary.normal,
    color: colors.palette.neutral250,
    fontSize: fontSizes[15],
    lineHeight: lineHeights[24],
    textAlign: "center",
  }

  const $tipText: TextStyle = {
    ...$headText,
    fontSize: fontSizes[9],
  }

  const $firstHeadText: TextStyle = {
    ...$headText,
    fontSize: fontSizes[12],
  }

  return {
    $headerContainer,
    $rowContainer,
    $userContainer,
    $addressText,
    $timeContainer,
    $outlined,
    $headText,
    $timeText,
    $tipText,
    $firstHeadText,
  }
}
