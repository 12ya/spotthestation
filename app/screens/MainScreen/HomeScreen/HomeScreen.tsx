import { observer } from "mobx-react-lite"
import React from "react"
import { ViewStyle } from "react-native"
import { Screen } from "../../../components"
import { colors } from "../../../theme"
import { useSafeAreaInsetsStyle } from "../../../utils/useSafeAreaInsetsStyle"
import { Globe } from "../components/Globe"

export const HomeScreen = observer(function HomeScreen() {
  const $topInset = useSafeAreaInsetsStyle(["top", "bottom"], "padding")

  return (
    <Screen preset="fixed" contentContainerStyle={$container} style={[$topInset, {backgroundColor: colors.palette.neutral900}]} statusBarStyle="light">
      <Globe />
    </Screen>
  )
})

const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.backgroundDark,
  justifyContent: "space-between"
}