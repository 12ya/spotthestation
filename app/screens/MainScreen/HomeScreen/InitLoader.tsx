import React from "react"
import { ViewStyle, View, TextStyle } from "react-native"
import { Text } from "../../../components"
import { fontSizes, lineHeights, scale, typography } from "../../../theme"
import { colors } from "../../../theme/colors"
import { normalizeHeight } from "../../../utils/normalizeHeight"

export function InitLoader() {
  return (
    <View 
      accessible
      accessibilityLabel="coach mark"
      accessibilityHint="coach mark"
      accessibilityRole="text"
      style={$modalBodyContainer}
    >
      <Text tx="homeScreen.initLoader.message" style={$title} />
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.buttonBlue,
  borderRadius: scale(16),
  alignItems: 'center',
  paddingVertical: 36,
  paddingHorizontal: 30,
  width: '100%',
  alignSelf: 'center',
  marginTop: normalizeHeight(.28)
}

const $title: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: fontSizes[24],
  lineHeight: lineHeights[29],
  color: colors.palette.neutral100,
  paddingBottom: 12,
  paddingTop: 18
}
