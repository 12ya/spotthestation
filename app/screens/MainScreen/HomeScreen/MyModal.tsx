import Modal from "react-native-modal"
import React from "react"
import { observer } from "mobx-react-lite"
import { useStores } from "../../../models"

function MyModal({ name, children, ...rest }) {
  const { currentModal, closeModal } = useStores()
  return (
    <Modal
      {...rest}
      isVisible={Boolean(
        currentModal && currentModal.name === name && currentModal.state === "open",
      )}
      onModalHide={closeModal}
    >
      {children}
    </Modal>
  )
}

export default observer(MyModal)
