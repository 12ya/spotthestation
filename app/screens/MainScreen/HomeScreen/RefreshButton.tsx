import { StyleFn, useStyles } from "../../../utils/useStyles"
import React, { useEffect, useState } from "react"
import {
  PressableProps,
  Animated,
  Easing,
  StyleProp,
  ViewStyle,
  TouchableOpacity,
} from "react-native"
import { Icon, Text } from "../../../components"
import { colors, typography } from "../../../theme"

export interface RefreshButtonProps {
  onPress: PressableProps["onPress"]
  inProgress: boolean
  containerStyle?: StyleProp<ViewStyle>
}

export function RefreshButton({ onPress, inProgress, containerStyle }: RefreshButtonProps) {
  const { $container, $text } = useStyles(styles)

  const [spinValue] = useState(new Animated.Value(0))
  const [loop] = useState(
    Animated.loop(
      Animated.timing(spinValue, {
        toValue: 1,
        duration: 3000,
        easing: Easing.linear,
        useNativeDriver: true,
      }),
    ),
  )

  useEffect(() => {
    if (!inProgress) {
      loop.stop()
      loop.reset()
    } else loop.start()
  }, [inProgress])

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ["0deg", "180deg"],
  })

  return (
    <TouchableOpacity
      disabled={inProgress}
      onPress={onPress}
      hitSlop={{ top: 10, left: 10, bottom: 10, right: 10 }}
      style={[$container, containerStyle]}
    >
      <Animated.View style={{ transform: [{ rotate: spin }] }}>
        <Icon icon="refresh" size={24} color={colors.palette.neutral450} />
      </Animated.View>
      <Text style={$text} tx="homeScreen.selectLocation.refresh" />
    </TouchableOpacity>
  )
}

const styles: StyleFn = ({ fontSizes }) => {
  const $container: ViewStyle = {
    flexDirection: "row",
  }

  const $text = {
    fontFamily: typography.primary.normal,
    fontSize: fontSizes[12],
    color: colors.palette.neutral450,
    marginLeft: 3,
  }

  return { $container, $text }
}
