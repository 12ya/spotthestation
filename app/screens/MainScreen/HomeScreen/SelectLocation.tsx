/* eslint-disable camelcase */
import React, { useEffect, useState } from "react"
import { ViewStyle, View, Pressable, PressableProps, TextStyle, ScrollView } from "react-native"

import { Accessory, Icon, Text, TextField } from "../../../components"
import { colors, typography } from "../../../theme"
import { ExpandContainer } from "../components/ExpandContainer"
import { ListItem } from "../components/ListItem"
import { useSafeAreaInsetsStyle } from "../../../utils/useSafeAreaInsetsStyle"
import { getCurrentLocation, getNearbyPlaces, getPlaces } from "../../../utils/geolocation"
import { LocationType } from "../../OnboardingScreen/SignupLocation"
import Snackbar from "react-native-snackbar"

export interface SelectLocationProps {
  /**
   * A function for closing modal.
   */
  onClose?: PressableProps["onPress"]
}

export function SelectLocation({ onClose }: SelectLocationProps) {
  const [isFocus, setIsFocus] = useState(false)
  const [textValue, setTextValue] = useState("")
  const [current, setCurrent] = useState<LocationType | null>(null)
  const [nearby, setNearby] = useState<LocationType[]>([])
  const [searchResult, setSearchResult] = useState<LocationType[]>([])

  const $marginTop = useSafeAreaInsetsStyle(["top"], "margin")
  const $paddingBottom = useSafeAreaInsetsStyle(["bottom"], "padding")

  const getLocations = async () => {
    const location = await getCurrentLocation()
    const res = await getNearbyPlaces(location.location, 100)
    setCurrent(location)
    setNearby(res)
  }

  const setPlaces = async (value: string) => {
    const location = await getPlaces(value)
    
    setSearchResult(location)
  }

  useEffect(() => {
    getLocations().catch((e: Error) => {
      Snackbar.show({
        text: e.message || "Some error occured",
        duration: Snackbar.LENGTH_SHORT,
      })
    })
  }, [])

  useEffect(() => {
    setPlaces(textValue).catch((e: Error) => {
      Snackbar.show({
        text: e.message || "Some error occured",
        duration: Snackbar.LENGTH_SHORT,
      })
    })
  }, [textValue])

  return (
    <View style={[$modalBodyContainer, $marginTop, $paddingBottom]}>
      <Pressable
        style={$close}
        onPress={onClose}
      >
        <Icon icon="x" color={colors.palette.neutral450} />
      </Pressable>
      <Text tx="homeScreen.selectLocation.title" style={$title} />
      <TextField
        value={textValue}
        autoCapitalize="none"
        autoCorrect={false}
        placeholderTx="homeScreen.selectLocation.inputPlaceholder"
        onChangeText={setTextValue}
        onFocus={() => setIsFocus(true)}
        onBlur={() => setIsFocus(false)}
        inputWrapperStyle={isFocus ? [$locations, $active] : $locations}
        renderLeftAccessory={({ style }) => (
          <Accessory 
            icon="search"
            color={colors.palette.neutral450}
            style={style}
          />
        )}
        renderRightAccessory={(({ style }) => isFocus && textValue ? (
          <Accessory 
            style={style}
            icon={"xCircle"}
            onPress={() => setTextValue("")}
          />
        ) : (
          <Accessory 
            style={style}
            icon={"currentLocation"}
          />
        ))}
      />
      <ScrollView style={$scrollContainer}>
        {(!isFocus && searchResult.length === 0) && (
          <>
            {Boolean(current) && <ExpandContainer title="homeScreen.selectLocation.current" expandble={false}>
              <ListItem icon="pin" title={current.title} subtitle={current.subtitle} selected={true} />
            </ExpandContainer>}
            <ExpandContainer title="homeScreen.selectLocation.saved" itemsCount={2}>
              <ListItem icon="pin" title="George Bush St, Times Square" subtitle="New York City, NY, 83957" />
              <ListItem icon="pin" title="Navy Garden" subtitle="San Diego, CA, 54813" />
            </ExpandContainer>
            {nearby.length > 0 && <ExpandContainer title="homeScreen.selectLocation.nearby" itemsCount={nearby.length} defaultValue={false}>
                {nearby.map((place: LocationType) => <ListItem key={place.title} icon="pin" title={place.title} subtitle={place.subtitle} />)}
              </ExpandContainer>
            }
          </>
        )}
        {(isFocus || searchResult.length > 0) && 
          <ExpandContainer title="homeScreen.selectLocation.search" itemsCount={searchResult.length} expandble={false}>
            {searchResult.map((place: LocationType) => <ListItem key={place.title} icon="pin" title={place.title} subtitle={place.subtitle} />)}
          </ExpandContainer>
        }
      </ScrollView>
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.neutral350,
  borderTopLeftRadius: 18,
  borderTopRightRadius: 18,
  flex: 1
}

const $scrollContainer: ViewStyle = { 
  flex: 1, 
  paddingHorizontal: 36,
  marginBottom: 24
}

const $close: ViewStyle = {
  position: "absolute",
  top: 0,
  right: 0,
  padding: 18
}

const $title: TextStyle = {
  marginTop: 36,
  marginBottom: 24,
  fontFamily: typography.primary?.normal,
  fontSize: 36,
  lineHeight: 44,
  color: colors.palette.neutral250,
  paddingHorizontal: 36,
}

const $locations: ViewStyle = {
  borderWidth: 1.5,
  borderColor: "transparent",
  borderRadius: 28,
  height: 56,
  backgroundColor: colors.palette.neutral550,
  overflow: "hidden",
  marginHorizontal: 36,
}

const $active: ViewStyle = {
  borderWidth: 1.5,
  borderColor: colors.palette.buttonBlue,
  backgroundColor: colors.palette.overlayBlue,
}
