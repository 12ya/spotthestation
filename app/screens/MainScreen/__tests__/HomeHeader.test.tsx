/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import React from "react"
import renderer from "react-test-renderer"
import { HomeHeader } from "../HomeScreen/HomeHeader"

it("renders correctly", () => {
  const tree = renderer
    .create(
      <HomeHeader
        user={{ firstName: "John", address: "Continental" }}
        onLocationPress={() => ({})}
        onSightingsPress={() => ({})}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
