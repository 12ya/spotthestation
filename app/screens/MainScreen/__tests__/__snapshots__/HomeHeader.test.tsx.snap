// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 5,
      }
    }
  >
    <View
      style={
        Object {
          "width": "85%",
        }
      }
    >
      <Text
        accessibilityHint="address"
        accessibilityLabel="address"
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Helvetica Neue",
              "fontSize": 29,
              "lineHeight": 37,
            },
          ]
        }
      >
        Continental
      </Text>
    </View>
    <View
      accessibilityHint="open select location modal"
      accessibilityLabel="pin button"
      accessibilityRole="imagebutton"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1E2022",
            "borderRadius": 143,
            "display": "flex",
            "height": 60,
            "justifyContent": "center",
            "overflow": "hidden",
            "position": "relative",
            "width": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1E2022",
                "borderRadius": 143,
                "display": "flex",
                "height": 60,
                "justifyContent": "center",
                "overflow": "hidden",
                "position": "relative",
                "width": 60,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <ViewManagerAdapter_ExpoBlurView
          proxiedProperties={
            Object {
              "intensity": 0,
              "tint": "default",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View>
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              Array [
                Object {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              Object {
                "height": 29,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#FFFFFF",
                "width": 29,
              }
            }
            tintColor="#FFFFFF"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 5,
      }
    }
  >
    <View
      accessibilityHint="open sightings modal"
      accessibilityLabel="next sighting"
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#1E2022",
            "borderRadius": 15,
            "borderWidth": 2,
            "padding": 8,
            "width": "48%",
          },
          Object {
            "borderColor": "#0B3D91",
          },
        ]
      }
    >
      <Text
        accessibilityHint="sighting header"
        accessibilityLabel="sighting header"
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#ADADAE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 20,
              "lineHeight": 23,
              "textAlign": "center",
            },
          ]
        }
      >
        homeScreen.header.firstTimeHead undefined
      </Text>
      <Text
        accessibilityHint="sighting"
        accessibilityLabel="sighting"
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#DEDDDE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 26,
              "lineHeight": 35,
              "textAlign": "center",
            },
          ]
        }
      />
      <Text
        accessibilityHint="timezone"
        accessibilityLabel="timezone"
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#ADADAE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 15,
              "lineHeight": 23,
              "textAlign": "center",
            },
          ]
        }
      />
    </View>
    <View
      accessibilityHint="countdown to next sighting"
      accessibilityLabel="countdown"
      accessibilityRole="text"
      accessible={true}
      style={
        Object {
          "backgroundColor": "#1E2022",
          "borderRadius": 15,
          "borderWidth": 2,
          "padding": 8,
          "width": "48%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#ADADAE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 20,
              "lineHeight": 23,
              "textAlign": "center",
            },
          ]
        }
      >
        homeScreen.header.secondTimeHead undefined
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#DEDDDE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 26,
              "lineHeight": 35,
              "textAlign": "center",
            },
          ]
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {},
            Array [
              Object {
                "fontSize": 23,
                "lineHeight": 35,
              },
              Object {
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "color": "#191015",
              },
            ],
            undefined,
            undefined,
            Object {
              "color": "#ADADAE",
              "fontFamily": "Helvetica Neue",
              "fontSize": 15,
              "lineHeight": 23,
              "textAlign": "center",
            },
          ]
        }
      >
        DD:HH:MM:SS
      </Text>
    </View>
  </View>
</View>
`;
